addon/serializers/application.js

```diff
@@ -51,182 +51,207 @@ export default DS.RESTSerializer.extend({
-              // This occurs when request was made without the include query param.
-              if ( 'Pointer' === item.__type ) {
-                items[index] = item.objectId;
-
-              } else {
-                // When items are objects we need to clean them and add them to the store.
-                // This occurs when request was made with the include query param.
-                delete item.__type;
-                delete item.className;
-                item.id = item.objectId;
-                delete item.objectId;
-                item.type = relationship.type;
-                serializer.normalizeAttributes( relationship.type, item );
-                serializer.normalizeRelationships( relationship.type, item );
-                store.push( relationship.type, item );
-              }
+        /*
+         * TODO: Find a better way to do this
+         * Here we set the links property to a serialized version
+         * of key and className. This info will be passed to
+         * adapter.findHasMany where we can deserialize to create
+         * the needed Parse query.
+         */
+        if (hash[key] && 'hasMany' === relationship.kind) {
+          if (!hash[key].__op && hash[key].__op !== 'AddRelation') {
+            if (!hash.links) {
+              hash.links = {};
+            }
+            hash.links[key] = JSON.stringify({
+              key: key,
+              className: hash[key].className
             });
           }
-        }
-      }
-    }, this );
-
-    this._super( type, hash );
-  },
 
-  serializeIntoHash: function( hash, type, snapshot, options ) {
-    Ember.merge( hash, this.serialize( snapshot, options ) );
+          delete hash[key].__type;
+          delete hash[key].className;
+          hash[key] = [];
+        }
+      }, this);
+    }
   },
 
-  serializeAttribute: function( snapshot, json, key, attribute ) {
+  serializeAttribute(record, json, key, attribute) {
     // These are Parse reserved properties and we won't send them.
-    if ( 'createdAt' === key ||
-         'updatedAt' === key ||
-         'emailVerified' === key ||
-         'sessionToken' === key
+    if ('createdAt' === key ||
+        'updatedAt' === key ||
+        'emailVerified' === key ||
+        'sessionToken' === key ||
+        'password' === key
     ) {
       delete json[key];
 
     } else {
-      this._super( snapshot, json, key, attribute );
+      this._super(record, json, key, attribute);
     }
   },
 
-  serializeBelongsTo: function( snapshot, json, relationship ) {
-    var key         = relationship.key,
-        belongsToId = snapshot.belongsTo(key, { id: true });
+  serializeBelongsTo(record, json, relationship) {
+    var key = relationship.key,
+        belongsTo = record.belongsTo(key);
+
+    if (belongsTo) {
+      // @TODO: Perhaps this is working around a bug in Ember-Data? Why should
+      // promises be returned here.
+      if (belongsTo instanceof DS.PromiseObject) {
```

Comment:
Yeah this seems not good. `record` should be a snapshot, not a record. snapshots should not have promises on them afaik